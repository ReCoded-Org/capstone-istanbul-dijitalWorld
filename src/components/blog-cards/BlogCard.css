.readMoreButton {
 position: absolute;
 left: 0%;
 right: 0%;
 top: 0%;
 bottom: 0%;
 background: #6C567C;
 box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
 border-radius: 25px;
}