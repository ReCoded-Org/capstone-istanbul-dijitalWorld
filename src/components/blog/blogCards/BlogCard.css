.readMoreButton {
  color: #fff;
  display: block;
  margin-top: 30px;
  padding: 0.8em 1.5em;
  background: #6c567c;
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.082);
  border-radius: 5em;
  border: none;
}

.readMoreButton:active {
  outline: none;
}

.readMoreButton:hover {
  text-decoration-color: none;
}

.blogContainer {
  text-align: justify;
  margin: 100px auto;
}

.blogTitle {
  margin-bottom: 1em;
  color: #6c567b;
  font-size: 30px;
}

.blogBody {
  color: #4a474d;
  font-size: 20px;
  line-height: 150%;
}

.roundImage {
  width: 18em;
  height: 18em;
  align-self: center;
}

hr {
  border-top: 5px dotted lightgray;
  width: 2%;
}

.blogCardBox {
  margin-left: 2em;
}
