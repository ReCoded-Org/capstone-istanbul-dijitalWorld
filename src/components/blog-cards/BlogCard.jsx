import React from 'react';
import { Col, Row, Container, Image } from 'react-bootstrap';
import blogArticles from './blog.json';
import './BlogCard.css';

export default function BlogCard() {
  function handleClick() {
    // history.push('/`${blog.id}`');
  }

  return blogArticles.map((blog) => {
    return (
      <Container key={blog.id} className="blogContainer">
        <Row>
          <Col xs={6} md={6} lg={8}>
            <h3 className="blogTitle">{blog.title}</h3>
            <span className="blogBody">{blog.body.substring(0, 250)}...</span>
            <button onClick={handleClick} className="readMoreButton">
              Read More
            </button>
          </Col>
          <Col xs={4} md={4} lg={3} className="imageContainer">
            <Image src={blog.imgUrl} className="roundImage" roundedCircle />
          </Col>
        </Row>
      </Container>
    );
  });
}
