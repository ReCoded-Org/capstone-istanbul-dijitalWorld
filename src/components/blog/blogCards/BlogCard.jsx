import React from 'react';
import SingleBlogCard from './SingleBlogCard';
import './BlogCard.css';

export default function BlogCard() {
  const [allPosts, setAllPosts] = React.useState([]);
  const dataUrl =
    'https://public-api.wordpress.com/wp/v2/sites/worldwithwomen30560421.wordpress.com/posts';

  const getAllPosts = async () => {
    const data = await fetch(dataUrl);
    const posts = await data.json();
    setAllPosts(posts);
  };

  React.useEffect(() => {
    getAllPosts();
  }, []);

  return allPosts.map((post, index) => <SingleBlogCard post={post} />);
}
