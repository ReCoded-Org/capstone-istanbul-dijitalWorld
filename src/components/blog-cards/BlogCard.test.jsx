import React from 'react';
import BlogCard from './BlogCard.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<BlogCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
