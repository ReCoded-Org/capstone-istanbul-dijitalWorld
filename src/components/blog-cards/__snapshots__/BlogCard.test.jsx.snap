// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="blogContainer container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 col-md-6 col-6"
      >
        <h3
          className="blogTitle"
        >
          My Intersectional Activism as A Black African Muslim Woman
        </h3>
        <span
          className="blogBody"
        >
          My identities as being a Black Muslim woman who is also African come in one. I experience discrimination in many unique ways: my skin colour, my gender, my religion and being African. Often, the notion of have these identities tend to equate to me be
          ...
        </span>
        <button
          className="readMoreButton"
          type="button"
        >
          Read More
        </button>
      </div>
      <div
        className="imageContainer col-lg-3 col-md-4 col-4"
      >
        <img
          className="roundImage rounded-circle"
          src="/images/safe_image.jpeg"
        />
      </div>
    </div>
  </div>,
  <div
    className="blogContainer container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 col-md-6 col-6"
      >
        <h3
          className="blogTitle"
        >
          An Open Letter to My Abuser
        </h3>
        <span
          className="blogBody"
        >
          It is hard to put into words all that you put me through. This letter is, I know, another vain attempt to master the chaos within me. But trying is always a better option than quitting, I suppose. Sometimes I wish you could go through the pain and st
          ...
        </span>
        <button
          className="readMoreButton"
          type="button"
        >
          Read More
        </button>
      </div>
      <div
        className="imageContainer col-lg-3 col-md-4 col-4"
      >
        <img
          className="roundImage rounded-circle"
          src="/images/leon-biss.jpg"
        />
      </div>
    </div>
  </div>,
  <div
    className="blogContainer container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 col-md-6 col-6"
      >
        <h3
          className="blogTitle"
        >
          5 Things I’ve Learned from Malala Yousafzai
        </h3>
        <span
          className="blogBody"
        >
          Today is Malala Yousafzai’s birthday. As an activist, advocate for girls’ education, champion of human rights and Nobel Peace Prize winner, Malala’s words and actions offer inspiration and hope to people all over the world. In 2012, a Taliban gunman 
          ...
        </span>
        <button
          className="readMoreButton"
          type="button"
        >
          Read More
        </button>
      </div>
      <div
        className="imageContainer col-lg-3 col-md-4 col-4"
      >
        <img
          className="roundImage rounded-circle"
          src="/images/malala.jpg"
        />
      </div>
    </div>
  </div>,
]
`;
